### PR TITLE
Tests added for fyle report approval view

### PR DESCRIPTION
**Pull request by vishalpandeyvip**
Code coverage of fyle_slack_app/fyle/report_approvals/views.py improved from 32% to 98%
Total code coverage 75%
Coverage report attested below:
 
Screenshot from 2022-11-17 14-51-19

### CODE REVIEW

**Comment by ashwin1111** [comment]
@vishalpandeyvip checks are failing, can you assign the PR for review once those are fixed?

File: .gitignore [hunk 1]
```diff
@@ -137,3 +137,5 @@ nginx.conf
 .idea/
 *.pyc
 test-reports/
+ngrok.log
+pytest-coverage.txt
```
Comment (NileshPant1999) [inline .gitignore:141]: new line

File: Dockerfile [hunk 2]
```diff
@@ -3,8 +3,8 @@ FROM python:3.8-slim
 ENV PYTHONDONTWRITEBYTECODE 1
 ENV PYTHONUNBUFFERED 1
 
-RUN apt-get update && \
-    apt-get install -y libpq-dev gcc python3.8-dev
+RUN apt-get update && apt-get -y install libpq-dev gcc && apt-get install git -y --no-install-recommends
```
Comment (NileshPant1999) [inline Dockerfile:6]: why we removed python3.8-dev

File: fyle_slack_service/settings.py [hunk 3]
```diff
@@ -225,6 +225,10 @@
 FYLE_CLIENT_SECRET = os.environ['FYLE_CLIENT_SECRET']
 FYLE_SLACK_APP_SEGMENT_KEY = os.environ['FYLE_SLACK_APP_SEGMENT_KEY']
 FYLE_BRANCHIO_BASE_URI = os.environ['FYLE_BRANCHIO_BASE_URI']
+FYLE_TOKEN_URI = os.environ['FYLE_TOKEN_URI']
+FYLE_BASE_URL = os.environ.get('FYLE_BASE_URL')
+FYLE_REFRESH_TOKEN = os.environ['FYLE_REFRESH_TOKEN']
+FYLE_SERVER_URL = os.environ['FYLE_SERVER_URL']
```
Comment (NileshPant1999) [inline fyle_slack_service/settings.py:231]: make sure to add this is fyle deploy too

File: pytest.ini [hunk 4]
```diff
@@ -0,0 +1,8 @@
+[pytest]
+DJANGO_SETTINGS_MODULE = fyle_slack_service.settings
+python_files = tests.py test_*.py *_tests.py
+addopts = -p no:warnings --strict-markers --no-migrations --reuse-db
+log_cli = 1
+log_cli_level = INFO
+log_cli_format = %(asctime)s [%(levelname)8s] %(name)s: %(message)s (%(filename)s:%(lineno)s)
+log_cli_date_format=%Y-%m-%d %H:%M:%S
```
Comment (NileshPant1999) [inline pytest.ini:8]: new line

File: tests/conftest.py [hunk 5]
```diff
@@ -56,3 +57,31 @@ def mock_fyle() -> Platform:
...
+
+    return fyle_connection
+
+@pytest.fixture()
+def report_approval_user(request, mocker, test_connection):
+    mocker_1 = mocker.patch('fyle_slack_app.fyle.utils.get_fyle_sdk_connection', return_value = test_connection)
+    user = mock.Mock(spec = User)
+    user.fyle_refresh_token = 'dummy-refresh-token'
+    return user
+
```
Comment (NileshPant1999) [inline tests/conftest.py:87]: remove extra lines

File: Dockerfile (hunk 2 above)
Comment (Shwetabhk) [reply Dockerfile:6]: This is ok right
`FROM python:3.8-slim`

we're anyway install pythong

File: fyle_slack_service/settings.py (hunk 3 above)
Comment (Shwetabhk) [reply fyle_slack_service/settings.py:231]: I don't think slack app is using these tokens, they might be only used in tests
@vishalpandeyvip lets discuss this in syncup today

**Review by Shwetabhk — APPROVED** [review]

**Pull request merged**